datadog_checks_base/datadog_checks/base/utils/db/statement_metrics.py

```diff
@@ -68,12 +69,14 @@ def compute_derivative_rows(self, rows, metrics, key):
             # whether a metric is submitted for the row during this run or not.
             new_cache[row_key] = row
 
-            prev = self._previous_statements.get(row_key)
-            if prev is None:
-                continue
-
             metric_columns = metrics & set(row.keys())
 
+            # A row missing from cache is interpreted as one with zero values for all columns
+            prev = self._previous_statements.get(row_key, {k: 0 for k in metric_columns})
+            # Only generate derivative rows when the cache has already been initialized
+            if not self._cache_initialized:
+                continue
```

Comment:
This can now be moved up directly below L70.

Reply:
👍